book_source/05_developer_guide/04_coding_practices/Coding_style.Rmd

```diff
@@ -71,13 +71,24 @@ When another package is required by a function or script, it can be called in th
 
-> The `Suggests` field uses the same syntax as `Depends` and lists packages that are not necessarily needed. This includes packages used only in examples, tests or vignettes (see Section 1.4 [Writing package vignettes], page 26), and packages loaded in the body of functions. E.g., suppose an example from package foo uses a dataset from package bar. Then it is not necessary to have bar use foo unless one wants to execute all the examples/tests/vignettes: it is useful to have bar, but not necessary. Version requirements can be specified, and will be used by R CMD check.
+Despite the name, most dependencies should *not* be listed in `Depends`. Using `Depends` increases PEcAn's resource footprint and has the same drawbacks listed above for whole-namespace `@import`, while also making it easy to create subtle bugs that appear to work during interactive test sessions but fail inside of scripts. As the [R extensions manual](https://cran.r-project.org/doc/manuals/R-exts.html#Package-Dependencies) explains, `Depends` is not for developer convenience (emphasis added):
 
+> Field ‘Depends’ should nowadays be used rarely, only for packages which are intended to be put on the search path to make their facilities **available to the end user (and not to the package itself)**."
 
+The `Suggests` field can be used to declare dependencies on packages that make your package more useful but are not completely essential. Again from the [R extensions manual](https://cran.r-project.org/doc/manuals/R-exts.html#Package-Dependencies):
+
+> The `Suggests` field uses the same syntax as `Depends` and lists packages that are not necessarily needed. This includes packages used only in examples, tests or vignettes (see [Writing package vignettes](https://cran.r-project.org/doc/manuals/R-exts.html#Writing-package-vignettes)), and packages loaded in the body of functions. E.g., suppose an example from package foo uses a dataset from package bar. Then it is not necessary to have bar use foo unless one wants to execute all the examples/tests/vignettes: it is useful to have bar, but not necessary. Version requirements can be specified, and will be used by R CMD check.
+
+In some of the PEcAn model interface packages, we push the definition of "not necessarily needed" by declaring the coupled model package in `Suggests` rather than `Imports`. This is a compromise to simplify installation of PEcAn for users who only plan to use one or a few models: For example, the `PEcAn.BIOCRO` package cannot do anything useful when the BioCro model is not installed, but it lists BioCro in Suggests because *PEcAn as a whole* can work without it. This way users can avoid the bother of installing BioCro if they only wanted to run, say, SIPNET.
```

Comment:
Not necessarily for this PR, but it might be good to mention here the circumstances under which packages in `Suggests` are or are not installed. For instance, `install.packages` has the `dependencies` argument, and by default installs `Depends`, `Imports`, and `LinkingTo`, but _not_ `Suggests`.  I'm pretty sure `devtools::install*` also defaults to this, based on its documentation. However, in PEcAn's Makefile, we currently have `dependencies = TRUE`, which forces `Suggests` to also be installed. It might actually be worth changing that, since Suggests packages often give us a lot of problems.

Reply:
>mention here the circumstances under which packages in Suggests are or are not installed

Good call. I was thinking I should be explicit that DESCRIPTION isn't magic (especially: `Imports:` declares your /intention/ to import from those packages, it doesn't do anything to the namespace), and I can try to work that in as well.



> we currently have dependencies = TRUE, which forces Suggests to also be installed. It might actually be worth changing that, 

R CMD check is pretty insistent that it wants all Suggests installed. I think it'd be /possible/ to work around that, but given how many of our suggests are in fact used for checking I don't know if we'd save any hassle. Worth a try, though.